external/esp_idf_port/esp_lwip/dhcpserver.c

```diff
@@ -0,0 +1,1362 @@
+static ip4_addr_t client_address_plus;
+
+static list_node *plist = NULL;
+static bool renew = false;
+
+static dhcps_lease_t dhcps_poll;
+static dhcps_time_t dhcps_lease_time = DHCPS_LEASE_TIME_DEF;	//minute
+static dhcps_offer_t dhcps_offer = 0xFF;
+static dhcps_offer_t dhcps_dns = 0x00;
+static dhcps_cb_t dhcps_cb;
+static void add_netif_dhcps(struct netif *iface, struct udp_pcb *dhcps_pcb);
+static struct udp_pcb *get_netif_dhcps(struct netif *iface);
+static void reset_netif_dhcps(struct netif *iface);
+/******************************************************************************
+ * FunctionName : dhcps_option_info
+ * Description  : get the DHCP message option info
+ * Parameters   : op_id -- DHCP message option id
+ *                opt_len -- DHCP message option length
+ * Returns      : DHCP message option addr
+*******************************************************************************/
+static void add_netif_dhcps(struct netif *iface, struct udp_pcb *dhcps_pcb)
+{
+	dhcps_handle[netifnum].interface = iface;
+	dhcps_handle[netifnum++].dhcps_pcb = dhcps_pcb;
+	if (netifnum >= TCPIP_ADAPTER_IF_MAX) {
+		DHCPS_LOG("dhcps: overflow\n");
+	}
+}
+
+static struct udp_pcb *get_netif_dhcps(struct netif *iface)
+{
+	for (int i = 0; i < TCPIP_ADAPTER_IF_MAX; i++) {
+		if (dhcps_handle[i].interface == iface) {
+			return dhcps_handle[i].dhcps_pcb;
+		}
+	}
+	return NULL;
+}
+
+static void reset_netif_dhcps(struct netif *iface)
+{
+	for (int i = 0; i < TCPIP_ADAPTER_IF_MAX; i++) {
+		if (dhcps_handle[i].interface == iface) {
+			dhcps_handle[i].dhcps_pcb = NULL;
+		}
+	}
+}
+
+void *dhcps_option_info(u8_t op_id, u32_t opt_len)
+{
+	void *option_arg = NULL;
+
+	switch (op_id) {
+	case IP_ADDRESS_LEASE_TIME:
+		if (opt_len == sizeof(dhcps_time_t)) {
+			option_arg = &dhcps_lease_time;
+		}
+
+		break;
+
+	case REQUESTED_IP_ADDRESS:
+		if (opt_len == sizeof(dhcps_lease_t)) {
+			option_arg = &dhcps_poll;
+		}
+
+		break;
+
+	case ROUTER_SOLICITATION_ADDRESS:
+		if (opt_len == sizeof(dhcps_offer_t)) {
+			option_arg = &dhcps_offer;
+		}
+
+		break;
+
+	case DOMAIN_NAME_SERVER:
+		if (opt_len == sizeof(dhcps_offer_t)) {
+			option_arg = &dhcps_dns;
+		}
+
+		break;
+
+	default:
+		break;
+	}
+
+	return option_arg;
+}
+
+/******************************************************************************
+ * FunctionName : dhcps_set_option_info
+ * Description  : set the DHCP message option info
+ * Parameters   : op_id -- DHCP message option id
+ *                opt_info -- DHCP message option info
+ *                opt_len -- DHCP message option length
+ * Returns      : none
+*******************************************************************************/
+void dhcps_set_option_info(u8_t op_id, void *opt_info, u32_t opt_len)
+{
+	if (opt_info == NULL) {
+		return;
+	}
+	switch (op_id) {
+	case IP_ADDRESS_LEASE_TIME:
+		if (opt_len == sizeof(dhcps_time_t)) {
+			dhcps_lease_time = *(dhcps_time_t *) opt_info;
```

Comment:
Empty space after (). please check all the lines below.

Reply:
done